server/command/command.go

```diff
@@ -0,0 +1,110 @@
+package command
+
+import (
+	"encoding/json"
+
+	"github.com/mattermost/mattermost-plugin-todo/server/todo"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type Command struct {
+	Args      *model.CommandArgs
+	Log       logger
+	pluginAPI pluginAPI
+	todoAPI   todoAPI
+}
+
+type ResponseType string
+
+// ResponseType values
+const (
+	ResponseTypeEphemeral    ResponseType = "ephemeral"
+	ResponseTypePost         ResponseType = "post"
+	ResponseTypeBotCustomDM  ResponseType = "bot_custom_dm"
+	ResponseTypeBotReplyPost ResponseType = "bot_reply_post"
+	ResponseTypeBotDM        ResponseType = "bot_dm"
+	ResponseTypePayload      ResponseType = "payload"
+)
+
+type Response struct {
+	Type      ResponseType
+	Message   string
+	Todo      string
+	UserID    string
+	ChannelID string
+	IssueID   string
+	PostID    string
+}
+
+type logger interface {
+	LogError(msg string, keyValuePairs ...interface{})
+}
+
+type pluginAPI interface {
+	GetUserByUsername(name string) (*model.User, *model.AppError)
```

Comment:
3/5 We should not be cascading AppError into our code

Reply:
Yep I missed that